Improve behaviour for playing from history

chme opened the pull request:
This is an attempt at improving the selection of a song from the "previously played" list. 

Remote sends the command "playnow" with mode "-1" if an entry of the history was selected. The index parameter is then the position of this entry in the history queue. Currently the mode parameter is not taken into account and therefor issuing this action results in playing the song from the "up next" queue at the position identified by the index parameter.

With these changes if mode is "-1", forked-daapd will look in the history to identify the song to start playing.
If the song is found and it is also present in the playqueue, forked-daapd will skip back to this song in the playqueue (gets the song id from the history and starts playing this id).

What is not working after these changes is, that if a song is not in the current playqueue, the queue will start from the beginning. Another problem is, if a song is in the playqueue more than once. forked-daapd will skip to the first occurrence in the playqueue.

ejurgensen commented on src/httpd_dacp.c:
```diff
@@ -783,12 +785,42 @@ dacp_reply_cue_play(struct evhttp_request *req, struct evbuffer *evbuf, char **u
+    {
+      /* If mode parameter is -1, the index is relative to the history queue, otherwise to the Up Next queue */
+      param = evhttp_find_header(query, "mode");
+      if (param && strcmp(param, "-1") == 0)
```
Could you add parenthesis, so: (strcmp(param, "-1") == 0)

ejurgensen commented on src/httpd_dacp.c:
```diff
@@ -783,12 +785,42 @@ dacp_reply_cue_play(struct evhttp_request *req, struct evbuffer *evbuf, char **u
+	  history = player_history_get();
+	  if (history->count > pos)
+	    {
+	      pos = history->id[(history->start_index + history->count - pos - 1) % MAX_HISTORY_COUNT];
```
Could you split this in two lines, so first calculate the id and then find the pos?

chme commented:
Updated and rebased. I have to admit that the code is really difficult to read and follow. But refactoring it will lead to a lot of changes with the potential of breaking things. But if you have some ideas how to improve things, i can try to do it.

ejurgensen commented:
Thanks for this. I don't really have any ideas for simplification, but yes, I have trouble following the code too.
